lib/index.coffee

```diff
@@ -10,13 +10,24 @@ dockerUtils = require './docker'
 
 class DockerGC
 	setDocker: (hostObj) ->
+		@baseImage = null
 		@currentMtimes = {}
 		@hostObj = _.defaults({ Promise }, hostObj)
 		@dockerProgress = new DockerProgress(@hostObj)
 		dockerUtils.getDocker(@hostObj)
 		.then (docker) =>
 			@docker = docker
 
+			# Docker info can take a while so do it here,
+			# and don't wait on the results
+			@getDaemonArchitecture()
+			.then (arch) =>
+				@baseImage = switch arch
+					when 'armv8l' then 'aarch64/alpine'
+					when 'armhf', 'armv7l' then 'armhf/alpine'
+					when 'x86_64' then 'alpine'
+			return
```

Comment:
This should be `return null` so that you don't get warnings about not returning the promise above